pkg/diskmaker/controllers/lvset/reconcile.go

```diff
@@ -0,0 +1,297 @@
+	lvset := &localv1alpha1.LocalVolumeSet{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, lvset)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// get node name
+	nodeName := getNodeNameEnvVar()
+	// get node labels
+	node := &corev1.Node{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: nodeName}, node)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// ignore LocalVolmeSets whose LabelSelector doesn't match this node
+	// NodeSelectorTerms.MatchExpressions are ORed
+	matches, err := nodeSelectorMatchesNodeLabels(node, lvset.Spec.NodeSelector)
+	if err != nil {
+		reqLogger.Error(err, "failed to match nodeSelector to node labels")
+		return reconcile.Result{}, err
+	}
+
+	if !matches {
+		return reconcile.Result{}, nil
+	}
+
+	storageClass := lvset.Spec.StorageClassName
+
+	// get associated provisioner config
+	cm := &corev1.ConfigMap{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: nodedaemon.ProvisionerConfigMapName, Namespace: request.Namespace}, cm)
+	if err != nil {
+		reqLogger.Error(err, "could not get provisioner configmap")
+		return reconcile.Result{}, err
+	}
+
+	// read provisioner config
+	provisionerConfig := staticProvisioner.ProvisionerConfiguration{}
+	staticProvisioner.ConfigMapDataToVolumeConfig(cm.Data, &provisionerConfig)
+
+	// get symlinkdir
+	symLinkConfig, ok := provisionerConfig.StorageClassConfig[storageClass]
+	if !ok {
+		return reconcile.Result{}, fmt.Errorf("could not find storageclass entry %q in provisioner config: %+v", storageClass, provisionerConfig)
+	}
+	symLinkDir := symLinkConfig.HostDir
+
+	// list block devices
+	blockDevices, badRows, err := internal.ListBlockDevices()
+	if err != nil {
+		r.eventReporter.Report(lvset, newDiskEvent(diskmaker.ErrorRunningBlockList, "failed to list block devices", "", corev1.EventTypeWarning))
+		reqLogger.Error(err, "could not list block devices", "lsblk.BadRows", badRows)
+		return reconcile.Result{}, err
+	} else if len(badRows) > 0 {
+		r.eventReporter.Report(lvset, newDiskEvent(diskmaker.ErrorRunningBlockList, fmt.Sprintf("error parsing rows: %+v", badRows), "", corev1.EventTypeWarning))
+		reqLogger.Error(fmt.Errorf("bad rows"), "could not parse all the lsblk rows", "lsblk.BadRows", badRows)
+	}
+
+	// find disks that match lvset filters and matchers
+	validDevices := r.getValidDevices(reqLogger, lvset, blockDevices)
+
+	// process valid devices
+	var noMatch []string
+	for _, blockDevice := range validDevices {
+		devLogger := reqLogger.WithValues("Device.Name", blockDevice.Name)
+
+		// validate MaxDeviceCount
+		var alreadyProvisionedCount int
+		alreadyProvisionedCount, noMatch, err = getAlreadyProvisioned(symLinkDir, validDevices)
```

Comment:
Can this use `GetMatchingSymlinksInDirs` function?

Reply:
It also reports nonMatching symlinks